src/webview/css/css.js

```diff
@@ -1,11 +1,58 @@
 /* @flow strict-local */
+import { Platform } from 'react-native';
 import type { ThemeName } from '../../types';
 import cssPygments from './cssPygments';
 import cssEmojis from './cssEmojis';
 import cssNight from './cssNight';
 
+/** CSS fragment to support scrolling of KaTeX formulae. */
+/*
+  By default, KaTeX renders (non-inline) math into a div of fixed precomputed
+  width. This will be cut off by the edge of the screen when the formula is too
+  long -- and on a mobile device, that's very nearly always.
+
+  The naïve solution of simply giving some part of the KaTeX fragment itself an
+  `overflow-x: auto` style breaks terribly:
+    * Margin collapsing no longer works, causing rendering artifacts. (This is
+      particularly visible on integral signs.)
+    * `overflow-y: hidden` isn't respected. If KaTeX has used negative-position
+      struts in its rendering (which it does frequently), there will always be
+      vertical scrollability. (This may be a Chrome bug.)
+
+  The added border functions as a UI hint to indicate that scrolling is
+  possible and necessary. (This may not otherwise be apparent, if the equation
+  is cut off between two symbols.)
+*/
+const katexScrollCss = `<style id="katex-mobile-scroll">
+.zulip-katex-outer { display: block; overflow-x: auto; }
+.zulip-katex-inner {
+  display: inline-block;
+  border: 1px solid hsla(187, 35%, 51%, .5);
+  border-radius: 0.25em;
+}
+.zulip-katex-inner .katex-display { margin: 0.5em 0.25em; }
```

Comment:
That's informative, thanks.

Thinking more about this, I really do want it to follow our style. (Policy? I said "style" originally but as I'm rereading what we have on this I realize that's inadequate -- it makes a visible difference, and overuse of `em` is an accessibility issue.) There are good reasons for that style^W policy, and we can't (within reasonable effort) control how third-party libraries work but we can keep things 100% consistent within our own code.

And in this case, to the extent there's a clash between our usage and the one in KaTeX's CSS, I don't think the result is materially worse if this rule is on our side of the clash rather than KaTeX's; I expect it works out better, for the same reasons that our usage is preferable in general.

I'm also finding as I look at this that the writeups we have of the reasoning here are a bit scattered and not entirely satisfying. So I'll go and pull those together into a better doc, which hopefully will be helpful.

Reply:
> I'm also finding as I look at this that the writeups we have of the reasoning here are a bit scattered and not entirely satisfying. So I'll go and pull those together into a better doc, which hopefully will be helpful.

Started that writeup as I entered this comment (just before the holiday), and went back and finished it just now: 78d57a28b or https://github.com/zulip/zulip-mobile/blob/master/docs/background/webview.md . PTAL!

Reply:
(Any followup discussion is probably best in a chat thread, rather than attempting to squeeze it in here.)